src/ophyd_async/epics/pmac/_pmac_trajectory.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+
+import numpy as np
+from scanspec.core import Path
+from scanspec.specs import Spec
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    FlyerController,
+    set_and_wait_for_value,
+    wait_for_value,
+)
+from ophyd_async.epics.motor import Motor
+from ophyd_async.epics.pmac import PmacIO
+from ophyd_async.epics.pmac._pmac_io import CS_LETTERS
+from ophyd_async.epics.pmac._utils import (
+    _PmacMotorInfo,
+    _Trajectory,
+    calculate_ramp_position_and_duration,
+)
+
+
+class PmacTrajectoryTriggerLogic(FlyerController):
+    def __init__(self, pmac: PmacIO) -> None:
+        self.pmac = pmac
+        self.scantime: float | None = None
+
+    async def prepare(self, value: Spec[Motor]):
+        slice = Path(value.calculate()).consume()
+        motor_info = await _PmacMotorInfo.from_motors(self.pmac, slice.axes())
+        ramp_up_pos, ramp_up_time = calculate_ramp_position_and_duration(
+            slice, motor_info, True
+        )
+        ramp_down_pos, ramp_down_time = calculate_ramp_position_and_duration(
+            slice, motor_info, False
+        )
+        trajectory = _Trajectory.from_slice(slice, ramp_up_time)
+        await asyncio.gather(
+            self._build_trajectory(
+                trajectory, motor_info, ramp_down_pos, ramp_down_time
+            ),
+            self._move_to_start(motor_info, ramp_up_pos),
+        )
+
+    async def kickoff(self):
+        if not self.scantime:
+            raise RuntimeError("Cannot kickoff. Must call prepare first.")
+        self.status = await self.pmac.trajectory.execute_profile.set(
+            True, timeout=self.scantime + 1
+        )
+
+    async def complete(self):
+        if not self.scantime:
+            raise RuntimeError("Cannot complete. Must call prepare first.")
+        await wait_for_value(
+            self.pmac.trajectory.execute_profile,
+            False,
+            timeout=self.scantime + DEFAULT_TIMEOUT,
+        )
+
+    async def stop(self):
+        await self.pmac.trajectory.abort_profile.set(True)
+
+    async def _build_trajectory(
+        self,
+        trajectory: _Trajectory,
+        motor_info: _PmacMotorInfo,
+        ramp_down_pos: dict[Motor, np.float64],
+        ramp_down_time: float,
+    ):
+        trajectory = trajectory.with_ramp_down(ramp_down_pos, ramp_down_time, 0)
```

Comment:
Should: Given you change this method to return self, the method name should probably be a verb, like:
```suggestion
        trajectory = trajectory.append_ramp_down(ramp_down_pos, ramp_down_time, 0)
```

Reply:
I think that's a good point and the name looks better as `append_ramp_down`!